### DISCUSSION TITLE
Change link (mark) text

### DISCUSSION THREAD

**Discussion by iamursky** [ueberdosis/tiptap#1869] [General Questions]
I'm trying to replace link text. I've tried `insertContent` and `insertContentAt` but they removes the mark itself:

```typescript
// Bad code: link gets removed
editor
  .chain()
  .focus()
  .extendMarkRange("link")
  .insertContent(text)
  .setLink({ href, target })
  .run();
```

How to change the text of a link?

**Comment by duhaime** [comment]
@hanspagel @philippkuehn is there any chance one of you might be able to give us an idea how we can change the text of a link?

**Comment by Harm-Nullix** [comment]
It is not that hard, but you need to wrap it in an a tag:

```
editor
              .chain()
              .focus()
              .extendMarkRange('link')
              .insertContent(`<a href="${href}">${text}</a>`)
              .setLink({ href: settings.url, target: '_blank' })
              .run()
```

The extension will handle the rest.
